File: leetcode/leetcode-js-solutions/plus-one/plus-one.js
const plusOne = (digits) => {
  for (let i=digits.length-1; i>=0; i--) {
    if (digits[i] != 9) {
      digits[i] += 1; 
      return digits; 
    } else {
      digits[i] = 0; 
    }
  }
  // edge case - if all digits are 9
  digits.unshift(1); 
  return digits; 
}

plusOne([9]); 

File: leetcode/leetcode-python-solutions/plus-one.py
def plus_one(digits):
    for i in range(len(digits)-1, -1, -1):
        if digits[i] != 9:
            digits[i] += 1
            return digits
        else:
            digits[i] = 0
    # edge case, if all digits are 9
    digits = [1] + digits
    print(digits)
    return digits


plus_one([9])
